### PR TITLE
Fix rules of the ClusterRole

### PR DESCRIPTION
**Pull request by hochuenw-dd**
## Proposed changes
Generated cluster rules are incorrect. 
Before:
Screen Shot 2020-11-23 at 7 36 51 PM
After:
Screen Shot 2020-11-23 at 11 18 56 PM

## Types of changes

What types of changes does your code introduce to Litmus? Put an `x` in the boxes that apply
- [ ] New feature (non-breaking change which adds functionality)
- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation Update (if none of the other choices applies)

## Checklist

Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code.
- [x] I have read the [CONTRIBUTING](https://github.com/litmuschaos/litmus/blob/master/CONTRIBUTING.md) doc
- [x] I have signed the commit for DCO to be passed.
- [ ] Lint and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works (if appropriate)
- [ ] I have added necessary documentation (if appropriate)

## Dependency
- Please add the links to the dependent PR need to be merged before this (if any).

### CODE REVIEW

File: docs/litmus-operator-ci.yaml [hunk 1]
```diff
@@ -32,11 +32,26 @@ metadata:
...
-  resources: ["pods","configmaps","events","services","chaosengines","chaosexperiments","chaosresults"]
+- apiGroups: [""]
+  resources: ["replicationcontrollers","secrets"]
+  verbs: ["get","list"]
+- apiGroups: ["apps.openshift.io"]
+  resources: ["deploymentconfigs"]
+  verbs: ["get","list"]
+- apiGroups: ["apps"]
+  resources: ["deployments", "daemonsets", "replicasets", "statefulsets"]
+  verbs: ["get","list"]
```
Comment (ispeakc0de) [inline docs/litmus-operator-ci.yaml:43]: deletecollection permission is required for deploymments, daemonsets, jobs and pods for abort case

Comment (ksatchit) [reply docs/litmus-operator-ci.yaml:43]: do we really need it for deployments and replicasets @ispeakc0de ? Just realised we need it for daemonsets to support the cases where POD_AFFECTED_PERC is > 0% wherever applicable. 

So I guess we separate out the rule for apps/daemonsets ?

Comment (ksatchit) [reply docs/litmus-operator-ci.yaml:43]: In fact, we don't need it on daemonsets either, realized we shifted to launching pods on multiple nodes over daemonsets. So changes are needed on job alone.

File: docs/litmus-operator-ci.yaml [hunk 2]
```diff
@@ -32,11 +32,26 @@ metadata:
...
+- apiGroups: ["batch"]
+  resources: ["jobs"]
+  verbs: ["get","list"]
```
Comment (ksatchit) [inline docs/litmus-operator-ci.yaml:46]:
```suggestion
  verbs: ["get","list","deletecollection"]
```

**Review by ksatchit — APPROVED** [review]

**Review by ispeakc0de — APPROVED** [review]

**Pull request merged**